### DISCUSSION TITLE
GlobalTransform component not synchronized in bevy 0.10 for entities spawned after TransformSystem::TransformPropagate?

### DISCUSSION THREAD

**Discussion by antilankin** [bevyengine/bevy#8077] [Q&A] [answered]
Hi all,

I'm working on a tile-based map editor for a game my brother is building, and I'm in the process of migrating it to bevy 0.10. While, on the whole, this works really well thanks to the excellent migration guide, I ran into a problem:

I'm running, among others, three systems:

In bevy 0.9, I ran them like this:
```
app
    .add_system_to_stage(
        CoreStage::PostUpdate,
        mouse_system
            .after(TransformSystem::TransformPropagate)
            .label("Selection")
    )
    .add_system_set_to_stage(
        CoreStage::PostUpdate,
        SystemSet::new()
            .after("Selection")
            .with_system(spawn_entity)
            // with other systems
    )
    .add_system_to_stage(
        CoreStage::Last,
        clear_selection_system
    );
```

In bevy 0.10, the systems look like this:
```
app
    .add_system(mouse_system.in_base_set(CoreSet::PostUpdate).after(TransformSystem::TransformPropagate))
    .add_system(spawn_entitiy.in_base_set(CoreSet::PostUpdate).after(mouse_system))
    // add other systems
    .add_system(clear_spawn_marker.in_base_set(CoreSet::Last));
```

The `mouse_system` adds a resource containing tile coordinates when the left button is clicked. I'm using `Camera::viewport_to_world`, that's why I scheduled the system after `TransformSystem::TransformPropagate`. The `spawn_entity` system looks for the resource and spawns an entity at the specified position. The `clear_spawn_marker` system removes the resource again.

While this worked in bevy 0.9, my spawned entities appear at the origin since the migration to 0.10. Some debugging revealed that while the `Transform` component shows the correct position, the `GlobalTransform` component of the spawned entities is `(0,0,0)`. 

I was expecting that the `GlobalTransform` would be updated on the next tick, but this isn't the case. While my issue is easily solved, for instance by re-arranging the systems, I'm curious where the difference between 0.9 and 0.10 is coming from.

**Comment by tim-blackbird** [comment, marked as answer]
It's likely because of the bug that's being fixed here: https://github.com/bevyengine/bevy/pull/8029

**Reply by antilankin** [reply]
Looks like that's it, thanks for the quick answer!
